packages/lisk-transaction-pool/src/transaction_pool.ts

```diff
@@ -20,16 +21,22 @@ export interface TransactionObject {
 	receivedAt?: Date;
 	readonly recipientId: string;
 	readonly senderPublicKey: string;
+	signatures?: ReadonlyArray<string>;
 	readonly type: number;
 }
 
 export interface TransactionFunctions {
 	containsUniqueData?(): boolean;
+	isExpired(date: Date): boolean;
```

Comment:
do we need to insert `date` here?

Reply:
I want to supply the current time (or the time to compare against) to this function so that every transaction is compared against the same time. Do you think it's unnecessary?

Reply:
I think comparing against the same time is a good point.
maybe make it as optional like `isExpired(date?: Date)`? so if date doesn't exist, transaction will create new Date by themself ?

Reply:
These interfaces are rudimentary, I think their names and parameters can be finalised after base Transaction PR is merged.